helix-core/src/main/java/org/apache/helix/controller/stages/CurrentStateComputationStage.java

```diff
@@ -356,6 +356,11 @@ void handleResourceCapacityCalculation(ClusterEvent event, ResourceControllerDat
       CurrentStateOutput currentStateOutput) {
     Map<String, Resource> resourceMap = event.getAttribute(AttributeName.RESOURCES.name());
     if (skipCapacityCalculation(cache, resourceMap, event)) {
+      // Ensure instance capacity is null if there are no resources. This prevents using a stale map when all resources
+      // are removed and then a new resource is added.
+      if (resourceMap == null || resourceMap.isEmpty()) {
```

Comment:
nit: can we merge the 2 if?

Reply:
I originally attempted to merge the two ifs as `if (resourceMap == null || resourceMap.isEmpty())` is basically a redundant check because the same check is made in `skipCapacityCalculation`
However, I was concerned with putting the responsibility of clearing the cache within skipCapacityCalculation because I did not want to give the method a side effect that may not be obvious to the user. Do you have any thoughts on how to approach this to simplify the logic?